src/ngraph/frontend/onnxifi/span.hpp

```diff
@@ -0,0 +1,104 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include <iterator>
+#include <stdexcept>
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        template <typename T>
+        class Span
```

Comment:
Do we really need this? I looks like a standard container. Can't we just use a standard container?
There are no comments describing what this is.
There are no unit tests for this.

Reply:
Please add a note about this as documentation.

Reply:
@apwojcik you can move your comment you just wrote into the header file as a comment. This is information which should have been in the header in the first place.
Does Span function identically to std::span? If so then you can put a c++ version check around it and put it in the std namespace under the class name `std::span`. That way when we switch to c++20 all of the code works the same.